src/error.rs

```diff
@@ -103,8 +103,8 @@ impl Display for Error {
         match self {
             Error::Status(status, response) => {
                 write!(f, "{}: status code {}", response.get_url(), status)?;
-                if let Some(original) = response.history().last() {
-                    write!(f, " (redirected from {})", original.get_url())?;
+                if let Some(original) = response.history().iter().last() {
```

Comment:
If you remove the `Hist` iterator, you'll want to change this `.last()` to `.first()`.